src/v/cloud_storage/remote_segment.h

```diff
@@ -299,6 +299,19 @@ class remote_segment final {
 
     std::optional<segment_chunks> _chunks_api;
     std::optional<offset_index::coarse_index_t> _coarse_index;
+
+    class consume_stream {
```

Comment:
nit: can we make the name a bit more descriptive? Maybe `chunk_caching_stream_consumer` or something?